Sources/Fullscreen/FullscreenGallery/GalleryPreviewView.swift

```diff
@@ -0,0 +1,170 @@
+        collectionView.clipsToBounds = false
+        return collectionView
+    }()
+
+    // MARK: - Init
+
+    init() {
+        super.init(frame: .zero)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    private func setup() {
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        clipsToBounds = false
+
+        addSubview(collectionView)
+        collectionView.fillInSuperview()
+
+        NSLayoutConstraint.activate([
+            heightAnchor.constraint(equalToConstant: cellSize.height)
+        ])
+
+    }
+
+    // MARK: - Public methods
+
+    public func superviewWillTransition(to size: CGSize) {
+        newSuperviewSize = size
+        collectionView.reloadSections(IndexSet(integer: 0))
+    }
+
+    public func scrollToItem(atIndex index: Int, animated: Bool) {
+        collectionView.scrollToItem(at: IndexPath(row: index, section: 0), at: .centeredHorizontally, animated: animated)
+    }
+
+    // MARK: - Private methods
+
+    private func reloadData() {
+        let imageCount = viewModel?.imageUrls.count ?? 0
+        images = [UIImage?](repeating: nil, count: imageCount)
+
+        collectionView.reloadData()
+
+        for counter in 0 ..< imageCount {
+            dataSource?.galleryPreviewView(self, loadImageWithWidth: cellSize.width, imageIndex: counter, dataCallback: { [weak self] (index, image) in
+                guard let self = self else { return }
+
+                if index >= 0 && index < self.images.count {
+                    self.images[index] = image
+                }
+
+                self.collectionView.reloadItems(at: [IndexPath(row: index, section: 0)])
+            })
+        }
+    }
+}
+
+extension GalleryPreviewView: UICollectionViewDataSource {
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return images.count
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeue(GalleryPreviewCell.self, for: indexPath)
+        let image = images[safe: indexPath.row]
+
+        cell.configure(withImage: image)
+
+        let estimatedFrame = CGRect(x: 0, y: 0, width: cellSize.width, height: cellSize.height)
+        cell.clipsToBounds = false
+        cell.layer.masksToBounds = false
+        cell.layer.shadowColor = UIColor.black.cgColor
+        cell.layer.shadowOpacity = 0.3
+        cell.layer.shadowOffset = CGSize(width: 0, height: 0)
+        cell.layer.shadowRadius = 3
+
+        cell.layer.shadowPath = UIBezierPath(rect: estimatedFrame).cgPath
+        cell.layer.shouldRasterize = true
+        cell.layer.rasterizationScale = UIScreen.main.scale
```

Comment:
Could this setup be placed into `GalleryPreviewCell.setup()` instead?

Reply:
The cell doesn't know his future size at all though, so it would have to be through a separate method on the form `GalleryPreviewCell.addShadow(withEstimatedSize: CGRect)`. What do you think of this?

Reply:
How about overriding `layoutSubviews` and set it there? I think the cell's frame should already be set.

Reply:
Ooh, that works! 🙂